common-auth/src/main/java/com/hortonworks/registries/auth/server/KerberosBasicAuthenticationHandler.java

```diff
@@ -0,0 +1,215 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.kerberos.authentication.KerberosAuthenticationProvider;
+import org.springframework.security.kerberos.authentication.sun.SunJaasKerberosClient;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Properties;
+
+/**
+ * The {@link KerberosBasicAuthenticationHandler} augments the Kerberos SPNEGO authentication mechanism with the Kerberos Basic Login authentication
+ * mechanism. If a user provides user credentials in a HTTPS, POST call, then a Kerberos login is attempted. In the authentication failure scenario,
+ * the SPNEGO sequence is invoked.
+ *
+ * <p>
+ * The supported configuration properties that are required in addition to those required by KerberosAuthenticationHandler are:
+ * <ul>
+ * <li>login.enabled: a boolean string to indicate whether the enabling of Kerberos login.</li>
+ * </ul>
+ */
+public class KerberosBasicAuthenticationHandler extends KerberosAuthenticationHandler {
+
+    public static final String LOGIN_ENABLED_CONFIG = "login.enabled";
+    public static final String SPNEGO_ENABLED_CONFIG = "spnego.enabled";
+    public static final String TYPE = "kerberos-login";
+    public static final String AUTHORIZATION_HEADER = "Authorization";
+    public static final String BASIC_AUTHENTICATION = "Basic";
+
+    private static final Logger LOG = LoggerFactory.getLogger(KerberosBasicAuthenticationHandler.class);
+    private static final String METHOD = "POST";
+
+    private KerberosAuthenticationProvider provider;
+    private boolean spnegoEnabled;
+
+    KerberosBasicAuthenticationHandler() {
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    public void init(Properties config) throws ServletException {
+        spnegoEnabled = Boolean.parseBoolean(config.getProperty(SPNEGO_ENABLED_CONFIG, Boolean.TRUE.toString()));
+        if (spnegoEnabled) {
+            super.init(config);
+        }
+
+        try {
+            provider = new KerberosAuthenticationProvider();
+            SunJaasKerberosClient client = new SunJaasKerberosClient();
+            client.setDebug(true);
```

Comment:
I am not sure how much logs the client will generate in debug mode. Can we enable this if debug mode is enabled for LOG?